docs_src/design/adr/security/0015-in-cluster-tls.md

```diff
@@ -0,0 +1,234 @@
+
+A microservice architecture normally strives for all of the above protections.
+
+Besides TLS, there are other mechanisms that can be used to provide some of the above properties.
+For example, IPSec tunnels provide confidentity, integrity, and authentication of the hosts (network-level protection).
+SSH tunnels provide confidentiality, integrity, and authentication of the tunnel endpoints (also network-level protection).
+TLS, however, is preferred, because it operates in-process at the application level and provides better point-to-point security.
+
+### Why to not encrypt?
+
+In the case of TLS communications,
+microservices depend on an asymmetric private key to prove their identity.
+To be of value, this private key must be kept secret.
+Applications typically depend on process-level isolation
+and/or file system protections for the private key.
+Moreover, interprocess communication using sockets is mediated by the
+operating system kernel.
+An attacker running at the privilege of the operating system
+has the ability to compromise TLS protections,
+such as by substituting a private key or certificate authority of their choice,
+accessing the unencrypted data in process memory,
+or intercepting the network communications that flow through the kernel.
+Therefore, within a single host, TLS protections may slow down an attacker,
+but are not likely to stop them.
+Additionally, use of TLS requires management of additional
+security assets in the form of TLS private keys.
+
+Microservice communcation across hosts, however,
+is vulnerable to intereception, and must be protected
+via some mechanism such as, but not limited to:
+IPSec or SSH tunnels, encrypted overlay networks,
+service mesh middlewares, or application-level TLS.
+
+Another reason to not encrypt is that TLS adds overhead to
+microservice communication in the form of additional network
+around-trips when opening connections
+and performing cryptographic public key and symmetric key operations.
+
+
+## Decision
+
+At this time, EdgeX is primarily a single-node IoT application framework.
+Should this position change, this ADR should be revisited.
+Based on the single-node assumption:
+
+- TLS will not be used for confidentiality and integrity of internal on-host microservice communication.
+- TLS will be avoided as an authentication mechanism of peer microservices.
+- Integrity and confidentiality of microservice communcations crossing host boundaries is required to secure EdgeX, but are an EdgeX customer responsibility.
+- EdgeX customers are welcome to add extra security to their own EdgeX deployments.
+
+
+## Consequences
+
+This ADR if approved would close the following issues as will-not-fix.
+
+- https://github.com/edgexfoundry/edgex-go/issues/1942
+- https://github.com/edgexfoundry/edgex-go/issues/1941
+- https://github.com/edgexfoundry/edgex-go/issues/2454
+- https://github.com/edgexfoundry/developer-scripts/issues/240
+- https://github.com/edgexfoundry/edgex-go/issues/2495
```

Comment:
I think part of this issue is solved by this ADR, however the title was just the tip of the iceberg. The issue also calls out a number of configuration naming inconsistencies that we hopefully can clean up as part of the Ireland release. That said, I'm going to go ahead and approve this ADR as is.